File: java-app/src/main/java/com/mercadolibre/java_app/services/shipping/ShippingOptionsDataSource.java
package com.mercadolibre.java_app.services.shipping;

import com.mercadolibre.java_app.models.shipping.FreeShippingOption;
import com.mercadolibre.java_app.models.shipping.NormalShippingOption;
import com.mercadolibre.java_app.models.shipping.ShippingOption;

import java.util.Arrays;
import java.util.List;

public final class ShippingOptionsDataSource {
    private ShippingOptionsDataSource() {}

    public static List<ShippingOption> retrieveShippingOptions()  {
        ShippingOption freeShippingOption = new FreeShippingOption("Gratis a domicilio");
        ShippingOption normalShippingOption = new NormalShippingOption(100.0);

        return Arrays.asList(freeShippingOption, normalShippingOption);
    }
}

File: kt-app/src/main/java/com/mercadolibre/melikotlinmeetup/services/shipping/ShippingOptionsDataSource.kt
package com.mercadolibre.melikotlinmeetup.services.shipping

import com.mercadolibre.melikotlinmeetup.commons.ShippingOption
import com.mercadolibre.melikotlinmeetup.models.shipping.FreeShippingOption
import com.mercadolibre.melikotlinmeetup.models.shipping.NormalShippingOption

fun retrieveShippingOptions(): List<ShippingOption> {
    val freeShippingOption = FreeShippingOption("Gratis a domicilio")
    val normalShippingOption = NormalShippingOption(100.0)

    return arrayListOf(freeShippingOption, normalShippingOption)
}